### PR TITLE
fix(link-wizard): persist saves even when isDirty misses a change

### PR DESCRIPTION
**Pull request by chiptus**
LinkWizardStep skipped the update mutation and silently advanced to the next artist whenever react-hook-form's `isDirty` flag was false, dropping the save with no error or toast. Removes that guard so Save & Next always persists the submitted data.

## Verification
- Open the Link Wizard, edit/select a provider link for an artist, click Save & Next; the artist's link is persisted and reflected on revisit.
- Added `LinkWizardStep.integration.test.tsx`: signs in as an admin test user, edits an artist's Spotify URL through the rendered wizard step, submits, and asserts the new URL was written to Supabase.

---
_Generated by [Claude Code](https://claude.ai/code/session_01ACt5WpGr1GTjm5atSYQ2St)_

### CODE REVIEW

File: src/pages/admin/festivals/LinkWizard/LinkWizardStep.integration.test.tsx [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+        onNext={onNext}
+      />,
+    );
+
+    const spotifyInput = await screen.findByLabelText("Spotify URL");
+    fireEvent.change(spotifyInput, { target: { value: newSpotifyUrl } });
+
+    fireEvent.click(screen.getByRole("button", { name: /save & next/i }));
+
+    await waitFor(() => expect(onNext).toHaveBeenCalled());
```
[the remark this answers is not in the document]
Comment (chiptus) [reply src/pages/admin/festivals/LinkWizard/LinkWizardStep.integration.test.tsx]: Good catch, confirmed by reverting the fix locally and re-running: the typed-input version passed either way. Rewrote the test to go through the candidate-select path (`form.setValue(..., { shouldDirty: true })` in `handleCandidateSelect`), which is the actual repro — with the old `isDirty` guard restored, this version fails (mutate never called); with the fix it passes. Also stubbed `useSearchArtistLinksQuery` so the test doesn't depend on the unserved search Edge Function in CI, per your other note. Pushed in 1261a79.

---
_Generated by [Claude Code](https://claude.ai/code)_
